src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -418,6 +416,79 @@ protected void WorkItemComplete()
             Test.Fixture = null;
         }
 
-#endregion
+        /// <summary>
+        /// Builds the set up tear down list.
+        /// </summary>
+        /// <param name="setUpMethods">Unsorted array of setup MethodInfos.</param>
+        /// <param name="tearDownMethods">Unsorted array of teardown MethodInfos.</param>
+        /// <returns>A list of SetUpTearDownItems</returns>
+        protected List<SetUpTearDownItem> BuildSetUpTearDownList(MethodInfo[] setUpMethods, MethodInfo[] tearDownMethods)
+        {
+            Guard.ArgumentNotNull(setUpMethods, nameof(setUpMethods));
+            Guard.ArgumentNotNull(tearDownMethods, nameof(tearDownMethods));
+
+            var list = new List<SetUpTearDownItem>();
+
+            Type fixtureType = Test.TypeInfo?.Type;
+            if (fixtureType == null)
+                return list;
+
+            while (fixtureType != null && !fixtureType.Equals(typeof(object)))
+            {
+                var node = BuildNode(fixtureType, setUpMethods, tearDownMethods);
+                if (node.HasMethods)
+                    list.Add(node);
+
+                fixtureType = fixtureType.GetTypeInfo().BaseType;
+            }
+
+            return list;
+        }
+
+        // This method builds a list of nodes that can be used to 
+        // run setup and teardown according to the NUnit specs.
+        // We need to execute setup and teardown methods one level
+        // at a time. However, we can't discover them by reflection
+        // one level at a time, because that would cause overridden
+        // methods to be called twice, once on the base class and
+        // once on the derived class.
+        // 
+        // For that reason, we start with a list of all setup and
+        // teardown methods, found using a single reflection call,
+        // and then descend through the inheritance hierarchy,
+        // adding each method to the appropriate level as we go.
```

Comment:
I implemented our own version of BindingFlags and many of the reflection APIs for portable, but only included BindingFlags that were in use in our code. That could be why DeclaredOnly isn't used, or it might go back to CF.